### PR TITLE
tests: Cleanup and add more GPU-AV Descriptor Indexing

### PR DESCRIPTION
**Pull request by spencer-lunarg**
Adds more end-to-end test for GPU-AV

### CODE REVIEW

File: tests/unit/gpu_av_descriptor_indexing_positive.cpp [hunk 1]
```diff
@@ -0,0 +1,445 @@
...
+
+    features2.features.robustBufferAccess = VK_FALSE;
+
+    RETURN_IF_SKIP(InitState(nullptr, &features2));
+}
+
+TEST_F(PositiveGpuAVDescriptorIndexing, Basic) {
+    TEST_DESCRIPTION("Basic indexing into a valid descriptor index");
+    RETURN_IF_SKIP(InitGpuVUDescriptorIndexing());
+    InitRenderTarget();
```
Comment (arno-lunarg) [inline tests/unit/gpu_av_descriptor_indexing_positive.cpp:49]: `InitRenderTarget` is probably not needed in the compute pipeline tests?

**Review by arno-lunarg — APPROVED** [review]
LGTM, one question
Nice to see more positive tests!

File: tests/unit/gpu_av_descriptor_indexing_positive.cpp [hunk 2]
```diff
@@ -0,0 +1,439 @@
...
+                              &descriptor_set.set_, 0, nullptr);
+    vk::CmdDispatch(m_commandBuffer->handle(), 1, 1, 1);
+    m_commandBuffer->end();
+
+    m_default_queue->submit(*m_commandBuffer);
+    m_default_queue->wait();
+}
+
+TEST_F(PositiveGpuAVDescriptorIndexing, UnInitImage) {
+    TEST_DESCRIPTION("Make sure there's not a crash if the sampler of a combined image sampler is initialized by the image isn't.");
```
Comment (filnet) [inline tests/unit/gpu_av_descriptor_indexing_positive.cpp:347]: typo: "by" should be "but" ?

Comment (spencer-lunarg) [reply tests/unit/gpu_av_descriptor_indexing_positive.cpp:347]: thanks, I have an another change with more GPU-AV tests coming, will fix it there

**Pull request merged**